src/pages/VaultsPage/VaultRow.tsx

```diff
@@ -34,17 +34,34 @@ export function VaultRow({
         setBalance(x.div(BigNumber.from(10).pow(vault.decimals)).toNumber());
       });
 
-  useBlockListener(updateBalance, [vault.id]);
+  const updateOwner = async () => {
+    const currentVault = contracts?.getVault(vault.vault_address);
+    if (!currentVault || !contracts) {
+      setUserIsOwner(false);
+      return;
+    }
+    const [ownerAddress, userAddress] = await Promise.all([
+      currentVault.owner(),
```

Comment:
even though updating the owner on every block is strictly speaking the correct thing to do, i think it's safe to assume it will change so infrequently (and can be handled by a page reload if it does) that it's a better use of resources to only query this once per vault.

Otherwise everything looks good here! 👍